src/value.rs

```diff
@@ -0,0 +1,134 @@
+// Copyright 2015 The Noise-rs Developers.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use num::Float;
+
+use {math, Seed};
+
+/// Linearly interpolate values.
+fn lerp<T: Float>(a: T, b: T, x: T) -> T {
+    a + x * (b - a)
+}
```

Comment:
Absolutely. I already have a longer interpolation function in it, so I can
just fold the two together.
On Jun 27, 2016 9:09 AM, "Brendan Zabarauskas" <notifications@github.com>
wrote:

> In src/value.rs
> <https://github.com/bjz/noise-rs/pull/109#discussion_r68572723>:
>
> > +//     http://www.apache.org/licenses/LICENSE-2.0
> > +//
> > +// Unless required by applicable law or agreed to in writing, software
> > +// distributed under the License is distributed on an "AS IS" BASIS,
> > +// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
> > +// See the License for the specific language governing permissions and
> > +// limitations under the License.
> > +
> > +use num::Float;
> > +
> > +use {math, Seed};
> > +
> > +/// Linearly interpolate values.
> > +fn lerp<T: Float>(a: T, b: T, x: T) -> T {
> > +    a + x * (b - a)
> > +}
>
> @Razaekel <https://github.com/Razaekel>: a candidate for your interp
> module?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/bjz/noise-rs/pull/109/files/75b2622978a491e76b20a3c2583236ba61279ba9#r68572723>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AEC6XhrwhVGGjlp_ingvBmFr9A_lI6LPks5qP8t4gaJpZM4I-iPE>
> .
>